src/DataTypes/IDataType.h

```diff
@@ -458,9 +503,11 @@ class IDataType : private boost::noncopyable
     /// This is mutable to allow setting custom name and serialization on `const IDataType` post construction.
     mutable DataTypeCustomNamePtr custom_name;
     mutable DataTypeCustomTextSerializationPtr custom_text_serialization;
+    mutable DataTypeCustomStreamsPtr custom_streams;
```

Comment:
Need to provide motivating example in comments.

Reply:
I've added comment to `IDataTypeCustomStreams`.